src/concurrency.md

```diff
@@ -70,7 +71,84 @@ fn main() {
 }
 ```
 
+[ex-rayon-thumbnails]: #ex-rayon-thumbnails
+<a name="ex-rayon-thumbnails"></a>
+## Generate jpg thumbnails in parallel
 
+[![rayon-badge]][rayon] [![glob-badge]][glob] [![image-badge]][image] [![cat-concurrency-badge]][cat-concurrency] [![cat-filesystem-badge]][cat-filesystem]
+
+This example uses the `glob`, `image`, and `rayon` crates to generate thumbnails for all .jpg files in the current directory. 
+The thumbnails are stored in a new folder called `thumbnails`.
+
+Files are found with [`glob::glob_with`] so we can match case insensitively on both `.jpg` and `.JPG`.
+
+```rust,no_run
+#[macro_use]
+extern crate error_chain;
+extern crate glob;
+extern crate image;
+extern crate rayon;
+
+use std::path::Path;
+use std::fs::{create_dir_all, File};
+
+use glob::{glob_with, MatchOptions};
+use image::{FilterType, ImageError};
+use rayon::prelude::*;
+
+error_chain! {
+    foreign_links {
+        Image(ImageError);
+        Io(std::io::Error);
+        Glob(glob::PatternError);
+    }
+}
+
+fn run() -> Result<()> {
+    // find all files in current directory that have a .jpg extension
+    // use the default MatchOptions so the search is case insensitive
+    let options: MatchOptions = Default::default();
+    let files: Vec<_> = glob_with("*.jpg", &options)?
+        .filter_map(|x| x.ok())
+        .collect();
+
+    let thumb_dir = Path::new("thumbnails");
+    create_dir_all(thumb_dir)?;
+
+    println!("Saving {} thumbnails into {:?}", files.len(), thumb_dir);
```

Comment:
instead of `{:?}` you can just call `thumb_dir.display()` and use standard formatting braces